src/configure/configure.ts

```diff
@@ -278,17 +278,17 @@ class Orchestrator {
         return this.inputs.potentialTemplates[0].targetType;
     }
 
-    private async analyzeNode(node: any): Promise<void> {
+    private async analyzeNode(node: IResourceNode | vscode.Uri): Promise<void> {
         if (!!node) {
-            if (await this.extractAzureResourceFromNode(node)) {
+            const folderNode = node as vscode.Uri;
+            if (folderNode && folderNode.fsPath) {
+                // right click on a folder
+                this.workspacePath = folderNode.fsPath;
+                telemetryHelper.setTelemetry(TelemetryKeys.SourceRepoLocation, SourceOptions.CurrentWorkspace);
+            } else if (await this.extractAzureResourceFromNode(node as IResourceNode)) {
```

Comment:
And here the `as` works without assigning first to a local variable?

Reply:
yeah, it is just passing the variable, so no issue.